### ISSUE TITLE
`--build-once` is broken in ReLaXed v0.2.5

### ISSUE THREAD

**Issue by domWalters** [domWalters/mkdocs-to-pdf#12]
https://github.com/RelaxedJS/ReLaXed/pull/196 broke the `--build-once` flag.

https://github.com/RelaxedJS/ReLaXed/pull/197 fixes it, but this has never been merged.

We should at least add a warning to the README.

**Comment by domWalters** [comment]
Having a look at the forks of ReLaXed, it looks like https://github.com/bgachenot/ReLaXed might be the best replacement to suggest.

**Comment by alessandroBalliniGPA** [comment]
Hi @domWalters , is your enhancement working?
how can i use you 0.9.4 preRelease?

**Comment by domWalters** [comment]
Hi,

---

If you just want to use the `master` branch of this repo:

```bash
pip install -e git://github.com/domWalters/mkdocs-to-pdf.git
```

---

If you need the `mermaid` fixes for mermaid diagrams:

```bash
pip install -e git://github.com/domWalters/mkdocs-to-pdf.git@mermaid-cli
```

For `mermaid` you also need the `mermaid-cli` tool available locally:

```bash
npm install -g @mermaid-js/mermaid-cli
```

---

I was intending on uploading this as a package to PyPi, but I've had a lot going on recently and changed jobs shortly after I made this fork.

If there'd be demand for a PyPi package, please let me know. You may want to open an issue that asks for one.

**Comment by alessandroBalliniGPA** [comment]
Installed mermaid-cli version but does not generate mermeid diagrams. It shows script instead of rendered images.
This my plugin section:

plugins:
    - mermaid2
    - with-pdf:
        render_js: true
        headless_chrome_path: /usr/bin/google-chrome-stable

**Comment by domWalters** [comment]
What does the output to the terminal say?

Are there any images generated in the output folder?

---

Hopefully I'll be able to look into this soon, but in the meantime for easier use:

* I've made a [`release-v0.9.4`](https://github.com/domWalters/mkdocs-to-pdf/tree/release-v0.9.4) branch.
* And a [`v0.9.4`](https://github.com/domWalters/mkdocs-to-pdf/releases/tag/v0.9.4) tag.

Just so any future master changes don't affect you.

---

Are you on a deadline that means this needs to be addressed quickly?

**Comment by alessandroBalliniGPA** [comment]
This is the output

[mkdocsOut.txt](https://github.com/domWalters/mkdocs-to-pdf/files/15365580/mkdocsOut.txt)

**Comment by domWalters** [comment]
Can you try disable the `mermaid2` plugin and see what happens?

Looking at that `mkdocsOut.txt` file, it seems like `mermaid2` is doing basically the same thing that `with_pdf` is trying to do.

In your case, I'd expect 7 lines of output saying "Converting mermaid diagram #" where # is a number starting at 0 ending at 6.

**Comment by domWalters** [comment]
Right, it looks like a newer `mkdocs` version changed the HTML for mermaid diagrams from:
```html
<pre class="mermaid"><code>...
```
...to...
```html
<pre><code class="language-mermaid">...
```

So the regex in `drivers/headless_chrome.py` misses the match.

This patch seems to work:

```diff
diff --git a/./drivers/headless_chrome.py b/home/dawalters/domWalters/mkdocs-to-pdf-mermaid/mkdocs_with_pdf/drivers/headless_chrome.py
index 6f4debb..ec97067 100644
--- a/./drivers/headless_chrome.py
+++ b/home/dawalters/domWalters/mkdocs-to-pdf-mermaid/mkdocs_with_pdf/drivers/headless_chrome.py
@@ -24,7 +24,7 @@ class HeadlessChromeDriver(object):

     def render(self, html: str, temporary_directory: pathlib.Path) -> str:
         try:
-            mermaid_regex = r'<pre><code class="language-mermaid">(.*?)</code></pre>'
+            mermaid_regex = r'<pre class="mermaid"><code>(.*?)</code></pre>'
             mermaid_matches = re.findall(mermaid_regex, html, flags=re.DOTALL)

             # Convert each Mermaid diagram to an image.
@@ -47,7 +47,7 @@ class HeadlessChromeDriver(object):

                 # Replace the Mermaid code with the image in the HTML string.
                 image_html = f'<img src="file://{image_file_path}" alt="Mermaid diagram {i+1}">'
-                html = html.replace(f'<pre><code class="language-mermaid">{mermaid_code}</code></pre>', image_html)
+                html = html.replace(f'<pre class="mermaid"><code>{mermaid_code}</code></pre>', image_html)

             self._logger.info(f"Post mermaid translation: {html}")
             with open(temporary_directory / "post_mermaid_translation.html", "wb") as temp:

```

**Comment by domWalters** [comment]
I just added this patch to the [`mermaid-cli`](https://github.com/domWalters/mkdocs-to-pdf/tree/mermaid-cli) branch.

I also upped the minimum `mkdocs` version to `v1.6.0` to match.

Can you try that and report back?

**Comment by alessandroBalliniGPA** [comment]
With mermaid2 activated:

![image](https://github.com/domWalters/mkdocs-to-pdf/assets/81438451/c5d95826-2b78-480f-9bd8-6d3e00aeb2f0)

without mermaid2 now there is something different:

![image](https://github.com/domWalters/mkdocs-to-pdf/assets/81438451/8819c200-2ad6-4016-9206-4d1a0af9c6e8)


i see some files .mmd in temp dir

![image](https://github.com/domWalters/mkdocs-to-pdf/assets/81438451/ce0b8857-fe29-4e31-b307-300c4a3c530b)

**Comment by domWalters** [comment]
Can you send me the terminal output?

The generated diagrams should be `.png` files that get made from the `.mmd` files by the `mmdc` (`mermaid-cli`) tool.

This would suggest to me that `mmdc` isn't available in the PATH where `mkdocs` is being executed.

**Comment by alessandroBalliniGPA** [comment]
That is:

![image](https://github.com/domWalters/mkdocs-to-pdf/assets/81438451/90f8fba0-89bc-4763-906b-632b241fdd22)

**Comment by domWalters** [comment]
Ah, your installed version of `node` is too old for the `mmdc` you have.

The newest version of `mmdc` needs node 16 or higher.

Not sure what your system setup is like, or how easy it would be for you to upgrade the system `node`, but you could look into using `nvm`:

https://github.com/nvm-sh/nvm

**Comment by alessandroBalliniGPA** [comment]
Wow, strange colors but now generated!

![image](https://github.com/domWalters/mkdocs-to-pdf/assets/81438451/c767ddcc-9676-411e-a596-2cc52222b08c)

**Comment by domWalters** [comment]
Currently, there is no way to set `mmdc` flags other than editing the code itself.

This is the line where the command is defined:

https://github.com/domWalters/mkdocs-to-pdf/blob/mermaid-cli/mkdocs_with_pdf/drivers/headless_chrome.py#L44

As you can see, `-t dark` is set but so is `-b transparent`. I think the original dev was prioritising printing the PDF, so wanted to avoid a large amount of colour.

It'd make sense for me to add a way to feed options through. I'm fairly busy for the rest of the weekend, but if I have some time I'll try add an option.
